### ISSUE TITLE
Surface sampling parameters in /v1/chat/completions (top_p, top_k, seed, stop, finish_reason, logprobs)

### ISSUE THREAD

**Issue by t8** [labscommunity/cascadia#14]
## Summary

The `/v1/chat/completions` endpoint accepts only `model`, `messages`, `max_tokens`, `temperature`, `stream` (`crates/tahoma-api/src/lib.rs:197-206`). Industry-standard fields are missing: `top_p`, `top_k`, `seed`, `stop`, `frequency_penalty`, `presence_penalty`, `logprobs`, `top_logprobs`. Streaming responses lack `usage` deltas, and `finish_reason` is always `"stop"` regardless of actual stop reason (`tahoma-api/src/lib.rs:417`).

`sparse-moe` already has `SamplingConfig` internally with `top_p` / `top_k` / `rep_penalty` / window. This issue surfaces those + several others at the HTTP API boundary and propagates them to all engines.

## Why

Tahoma's API hands non-default OpenAI clients an incomplete experience. Many evals (lm-eval-harness, llm-foundry, simple-evals) and downstream tools (LangChain, LlamaIndex, dspy, OpenAI Agents SDK) set these fields and expect them honored. Today they're silently dropped — clients have no way to tell that `top_p=0.9` they sent had zero effect.

`finish_reason` differentiation matters for observability — distinguishing "model finished naturally" (`stop`), "hit max_tokens" (`length`), "EOS token emitted" (also `stop` per spec), "explicit stop string matched" (`stop`), "client cancelled" (custom or `null`) is signal for routing decisions, retries, and quality checks.

### (1) Extend `ChatCompletionRequest`

**File:** `crates/tahoma-api/src/lib.rs:197-206`

Add fields:
```rust
pub struct ChatCompletionRequest {
    pub model: String,
    pub messages: Vec<ChatMessage>,
    #[serde(default)]
    pub max_tokens: Option<u32>,
    #[serde(default)]
    pub temperature: Option<f32>,
    #[serde(default)]
    pub top_p: Option<f32>,
    #[serde(default)]
    pub top_k: Option<u32>,
    #[serde(default)]
    pub seed: Option<u64>,
    #[serde(default)]
    pub stop: Option<StopSpec>, // string or array of strings, OAI-style
    #[serde(default)]
    pub frequency_penalty: Option<f32>,
    #[serde(default)]
    pub presence_penalty: Option<f32>,
    #[serde(default)]
    pub logprobs: Option<bool>,
    #[serde(default)]
    pub top_logprobs: Option<u32>,
    #[serde(default)]
    pub stream: Option<bool>,
    #[serde(default)]
    pub stream_options: Option<StreamOptions>,
}

#[derive(Deserialize)]
#[serde(untagged)]
pub enum StopSpec {
    Single(String),
    Multiple(Vec<String>),
}

#[derive(Deserialize)]
pub struct StreamOptions {
    #[serde(default)]
    pub include_usage: bool,
}
```

### (2) Extend the internal sampling type

**File:** `crates/tahoma-engine-sparse-moe/src/sampling.rs`

`SamplingConfig` exists. Add `frequency_penalty: f32`, `presence_penalty: f32`, `seed: Option<u64>`, `stop: Vec<String>`. The first two apply during logit shaping (count token occurrences in the current generation, subtract penalty); `seed` initializes the RNG before sampling; `stop` is matched against the running detokenized text after each step.

For the multi-stage Forward frame: `SAMPLING_WIRE_BYTES` will need to grow to accommodate the new fields. **Bump the frame-kind version** (currently `0x53_4D_45_02` Forward) to `0x03` so old peers fail fast at `parse_kind` rather than mis-parsing. See `crates/tahoma-engine-sparse-moe/src/dist.rs:30` for the frame versioning convention.

### (3) Route sampling through every engine

| Engine | Path | Notes |
|---|---|---|
| `sparse-moe` | `crates/tahoma-engine-sparse-moe/src/sampling.rs` + `runner.rs` | Already has the structure |
| `ov-genai` | `crates/tahoma-engine-openvino/src/genai.rs` | OV-GenAI's `GenerationConfig` already supports all these — translate directly |
| `ov-runtime` | `crates/tahoma-engine-openvino/src/runtime.rs` | Logits applied client-side; implement sampler there |
| `ov-dist-spec` | `crates/tahoma-engine-openvino/src/dist_spec.rs` | Same as runtime |

### (4) `finish_reason` differentiation

**File:** `crates/tahoma-types/src/lib.rs` (or wherever `Chunk` is defined)

Add a `FinishReason` enum:
```rust
#[derive(Clone, Copy, Debug, Serialize, Deserialize)]
#[serde(rename_all = "snake_case")]
pub enum FinishReason {
    Stop,        // EOS token or stop-string match
    Length,      // max_tokens hit
    Cancelled,   // client disconnect / explicit /v1/cancel
}
```

Have each engine emit the right one in the final chunk. API converts to OpenAI's string form (`stop`, `length`, `null` for in-progress streaming).

Current behavior: `tahoma-api/src/lib.rs:417` always emits `"stop"`. Change to read from `Chunk::finish_reason`.

### (5) Streaming usage deltas

OpenAI's SSE stream emits a final chunk with `usage: { prompt_tokens, completion_tokens, total_tokens }` when the client sets `stream_options.include_usage: true`. Implement same — Tahoma already has `prompt_tokens` and emits per-token chunks, so summation is trivial.

### (6) logprobs

OV-GenAI's `GenerationConfig` has a `logprobs` field; sparse-MoE samples logits locally and has the data on hand.

- `ChatCompletionRequest.logprobs` → engine generation config
- Engine emits `Chunk` with `logprobs: Some(Vec<TokenLogprob>)`
- API serializes per OpenAI schema: `{ "content": [ { "token": "...", "logprob": -1.23, "bytes": [...], "top_logprobs": [...] } ] }`

**Note:** `Chunk::logprobs` is currently `Option<()>` (a placeholder per the codebase audit). Define the real type.

Skip `logit_bias` for this issue — adds tokenizer complexity. Separate issue.

## Acceptance criteria

- [ ] `/v1/chat/completions` accepts all listed fields without 400 errors
- [ ] `top_p`, `top_k`, `seed`, `stop`, `frequency_penalty`, `presence_penalty` are honored by the `sparse-moe` engine. Smoke test: `stop: ["Berlin"]` truncates "Paris. The capital of Germany is Berlin" before "Berlin"
- [ ] Same sampling fields honored by `ov-genai` (passed through to `LLMPipeline.GenerationConfig`)
- [ ] `finish_reason` is `"length"` when `max_tokens` is hit, `"stop"` when EOS or stop-string is matched. Smoke test: `max_tokens: 5` returns `finish_reason: "length"`
- [ ] `logprobs: true` returns per-token logprobs in the response (OAI schema), at minimum for the `ov-genai` and `sparse-moe` engines
- [ ] `stream: true` with `stream_options: { include_usage: true }` emits a final SSE chunk containing `usage`
- [ ] Multi-stage pipeline-parallel respects the new sampling fields (verified by 2-box smoke test)
- [ ] Sampling-frame version bumped to `0x53_4D_45_03` (or whatever the next free low byte is); old peers reject the new frame with a clear error at `parse_kind`

## Effort

~ 2–3 days. logprobs alone is ~1 day; the rest is wiring + multi-stage frame bump + test coverage.

## Out of scope

- `logit_bias` — needs tokenizer integration; separate issue
- `tool_choice` / `tools` (function calling) — much larger work
- `response_format: json_schema` / grammar-constrained sampling — separate larger work
- `audio` / multimodal — out of scope

## References

- [OpenAI Chat Completions API reference](https://platform.openai.com/docs/api-reference/chat/create)
- [OpenVINO GenAI GenerationConfig](https://openvinotoolkit.github.io/openvino.genai/docs/structov_1_1genai_1_1GenerationConfig.html)
- Existing sparse-MoE SamplingConfig: `crates/tahoma-engine-sparse-moe/src/sampling.rs`
- Frame-kind versioning convention: `crates/tahoma-engine-sparse-moe/src/dist.rs:30`
